src/bvh/child_distance_traverse.rs

```diff
@@ -0,0 +1,336 @@
+                }
+                RestChild::None => {
+                    self.has_node = false;
+                }
+            },
+            BvhNode::Leaf { .. } => {
+                self.has_node = false;
+            }
+        }
+    }
+}
+
+impl<'shape, T, const D: usize, Shape: Bounded<T, D>, const ASCENDING: bool> Iterator
+    for ChildDistanceTraverseIterator<'_, 'shape, T, D, Shape, ASCENDING>
+where
+    T: BHValue,
+{
+    type Item = &'shape Shape;
+
+    fn next(&mut self) -> Option<&'shape Shape> {
+        loop {
+            if self.is_stack_empty() && !self.has_node {
+                // Completed traversal.
+                break;
+            }
+
+            if self.has_node {
+                // If we have any node, attempt to move to its nearest child.
+                let stack_info = self.move_first_priority();
+                // Save current node and farthest child
+                self.stack_push(stack_info)
+            } else {
+                // Go back up the stack and see if a node or leaf was pushed.
+                let (node_index, rest_child) = self.stack_pop();
+                self.node_index = node_index;
+                match self.bvh.nodes[self.node_index] {
+                    BvhNode::Node { .. } => {
+                        // If a node was pushed, now move to `unvisited` rest child, next in order.
+                        self.move_rest(rest_child);
+                    }
+                    BvhNode::Leaf { shape_index, .. } => {
+                        // We previously pushed a leaf node. This is the "visit" of the in-order traverse.
+                        // Next time we call `next()` we try to pop the stack again.
+                        return Some(&self.shapes[shape_index]);
+                    }
+                }
+            }
+        }
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::aabb::Bounded;
+    use crate::bvh::Bvh;
+    use crate::ray::Ray;
+    use crate::testbase::{generate_aligned_boxes, TBvh3, TPoint3, TVector3, UnitBox};
+    use std::collections::HashSet;
+
+    /// Create a `Bvh` for a fixed scene structure.
+    pub fn build_some_bvh() -> (Vec<UnitBox>, TBvh3) {
+        let mut boxes = generate_aligned_boxes();
+        let bvh = Bvh::build(&mut boxes);
+        (boxes, bvh)
+    }
+
+    /// Create a `Bvh` for an empty scene structure.
+    pub fn build_empty_bvh() -> (Vec<UnitBox>, TBvh3) {
+        let mut boxes = Vec::new();
+        let bvh = Bvh::build(&mut boxes);
+        (boxes, bvh)
+    }
+
+    fn traverse_distance_and_verify_order(
+        ray_origin: TPoint3,
+        ray_direction: TVector3,
+        all_shapes: &[UnitBox],
+        bvh: &TBvh3,
+        expected_shapes: &HashSet<i32>,
+    ) {
+        let ray = Ray::new(ray_origin, ray_direction);
+        let near_it = bvh.nearest_traverse_iterator(&ray, all_shapes);
+        let far_it = bvh.farthest_traverse_iterator(&ray, all_shapes);
```

Comment:
Should change old tests to point to the old iterator under the new name, not the new iterator under the old name.